src/mono/mono/eventpipe/ep-rt-mono.h

```diff
@@ -1796,6 +1806,28 @@ ep_rt_diagnostics_command_line_get (void)
 	return cmd_line;
 }
 
+static
+const ep_char16_t *
+ep_rt_entrypoint_assembly_path_get_ref_utf16 (void)
+{
+	// TODO: find mono entrypoint assembly path
+	return (const ep_char16_t *)L"TODO";
+}
+
+static
+const ep_char16_t *
+runtime_version_string_lazy_get_ref_utf16 (void)
+{
+	return (const ep_char16_t *)PRODUCT_VERSION_L;
+}
+
+static
+const ep_char16_t *
+ep_rt_runtime_version_get_ref_utf16 (void)
```

Comment:
if you switch to UTF8 for these API's you can use:
static
const ep_char8_t *
ep_rt_runtime_version_get (void)
{
    return EGLIB_TOSTRING (RuntimeProductVersion);
}

Reply:
I'd rather avoid needing to have to the extra lazy initialization for the converted utf16 string. Is it imperative that this method return a utf8 string?

Reply:
We can do any way we like, just that existing code mainly trying to returns UTF8. If you use the RuntimeProductVersion there is no need to convert it to utf16 in library, just return it using something similar to above and then let caller do the conversion to UTF16 when needed, similar to how its been done for command line, OS and arch specific details in existing process_protocol_helper_get_process_info.

If you still would like to handle these string differently, you will end up doing lazy init in Mono library for the main assembler path since its internally handled as UTF8.